props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+          }
+        return true;
+
+#else  // else BC_DUAL_BLADES is defined
+
+// -------------------- Any btn dual blades
+
+// Turn Main Blade ON
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_OFF):
+        TurnBladeOn(BC_MAIN_BLADE_SET);
+        return true;
+
+// Turn Main Blade ON additionally
+// Blaster Deflection
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_SAVED_CLICK_SHORT, MODE_ON):
+        if (!isMainBladeOn() && !on_pending_) {
+          TurnBladeOn(BC_MAIN_BLADE_SET);
+        } else {
+          DoBlasterBlock();
         }
-      }
-      return true;
+        return true;
 
-// Lightning Block
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_HELD_MEDIUM, MODE_ON):
-      //Don't lightning block if in colorchange mode
-      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
-      if (SaberBase::IsOn()) { // prevents triggering during preon
-        SaberBase::SetLockup(SaberBase::LOCKUP_LIGHTNING_BLOCK);
-        SaberBase::DoBeginLockup();
+// Turn Main Blade OFF
+      case EVENTID(BUTTON_POWER, EVENT_FIRST_HELD_MEDIUM, MODE_ON):
+        if (isMainBladeOn()) {
+          TurnBladeOff(BC_MAIN_BLADE_SET);
+        }
+        return true;
+
+// Thrust ON either non-ignited blade
+// Stab
+      case EVENTID(BUTTON_NONE, EVENT_THRUST, MODE_ON):
+        if (millis() - last_thrust_millis_ < 200 ||
+            millis() - saber_on_time_ < 2000) return false;        
+        // Check if both blades are already on
+        if (isMainBladeOn() && isSecondBladeOn()) {
+          if (on_pending_) return false;
+          SaberBase::SetClashStrength(2.0);
+          PVLOG_NORMAL << "** Doing STAB on " << (thrusting_blade_[BC_MAIN_BLADE] ? "MAIN blade.\n" : "SECOND blade.\n");
+          SaberBase::DoEffect(EFFECT_STAB, thrust_location);
+        } else {
+          if (thrusting_blade_ == BC_MAIN_BLADE_SET && isSecondBladeOn()) {
+            TurnBladeOn(BC_MAIN_BLADE_SET);
+            PVLOG_NORMAL << "** Main Blade THRUST ON\n";
+          } else if (thrusting_blade_ == BC_SECOND_BLADE_SET && isMainBladeOn()) {
+            TurnBladeOn(BC_SECOND_BLADE_SET);
+            PVLOG_NORMAL << "** Second Blade THRUST ON\n";
+          }
+        }
+        last_thrust_millis_ = millis();
+        return true;
+
+// Melt
+      case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_ON):
+        if (on_pending_) return false;
+        clash_impact_millis_ = millis();
+        if (!SaberBase::Lockup() && !swinging_) {
+          SaberBase::SetLockup(SaberBase::LOCKUP_MELT, forward_stab_ ? BC_MAIN_BLADE_SET : BC_SECOND_BLADE_SET);
+          PVLOG_NORMAL << "** Doing MELT on " << (forward_stab_ ? "MAIN blade.\n" : "SECOND blade.\n");
+          auto_melt_on_ = true;
+          SaberBase::DoBeginLockup();
+          }
+        return true;
+
+#endif  // BC_DUAL_BLADES
+
+// -------------------- Any btn any blades
+
+// Gesture Ignition Controls
+#ifdef BC_SWING_ON
```

Comment:
The next chunk of code is hard to read. There are just too many ifdefs.
I have some suggestions below for how to reduce it, but there are other ways, and you don't have to do all of them.

Reply:
fixed.